### ISSUE TITLE
[Urgent] Clarify in setting what happens to the API key

### ISSUE THREAD

**Issue by logankilpatrick** [reworkd/AgentGPT#24]
Right now, it is unclear where my API key is going after I copy and paste it into the page. Is it stored in browser? Deleted right away? This should be front and center in the settings page.

**Comment by asim-shrestha** [comment]
Hey good call, at work right now so I can't actually make changes but will tackle later today. User provided API keys are not stored anywhere long term (they just live in react state). They are passed directly into our backend requests and used directly.

Custom API key state: https://github.com/reworkd/AgentGPT/blob/main/src/pages/index.tsx#L22 (Filled in by the dialog, passed into AutonomousAgent to be used in requests)

Backend routes, each having a single param for the API key: https://github.com/reworkd/AgentGPT/tree/main/src/pages/api

**Comment by mrayonnaise** [comment]
Yeah, I walked through the code and noted this. We also probably need a clientEnv key that indicates that a key was provided server-side to prevent it from asking for one if not needed.

**Comment by asim-shrestha** [comment]
Thanks @mrayonnaise! Going to work on that confusing messaging now
